src/lib.rs

```diff
@@ -539,6 +539,31 @@ impl<T> MiniVec<T> {
     make_drain_filter_iterator(self, pred)
   }
 
+
+  #[inline]
+  #[must_use]
+  /// Drain method which returns new instance of `MiniVec`, created by moving content out of `self`.
+  ///
+  /// Comparing to `drain` method, this is just simple swap of pointers.
+  ///
+  /// As result any pointer to `self` becomes invalid.
+  ///
+  /// ```
+  /// use minivec::mini_vec;
+  ///
+  /// let mut vec = mini_vec![1, 2, 3, 4, 5, 6, 7, 9];
+  /// let new_vec = vec.drain_vec();
+  ///
+  /// assert_eq!(vec.len(), 0);
+  ///
+  /// assert_eq!(new_vec, [1, 2, 3, 4, 5, 6, 7, 9]);
+  /// ```
+  pub fn drain_vec(&mut self) -> Self {
+      let mut result = Self::new();
+      core::mem::swap(&mut result, self);
```

Comment:
I don't think this will work when `self` is default-constructed.

Default-constructed `MiniVec`s point to the same object.

`swap` seems to assume the two addresses aren't the same:
https://doc.rust-lang.org/stable/src/core/mem/mod.rs.html#685-691

You can see the impl of `swap_nonoverlapping_one` here:
https://doc.rust-lang.org/src/core/ptr/mod.rs.html#431

I think we need to either special-handle the default case.

Reply:
I'm not sure I follow you here.
When you perform swap you take copy of one element, write into it another one and then write copy.
`swap` assumes that two addresses are not the same because you cannot create two mutable references to the same variable.

So I think you misunderstood our case, addresses of `self` and `result` cannot be the same otherwise it would be UB as we created two mutable references to the same value.
But these are two different values :)